core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/rename.kt

```diff
@@ -23,12 +29,104 @@ import kotlin.reflect.KProperty
 
 // region DataFrame
 
+/**
+ * Renames the specified [columns\] keeping their original values and location within the [DataFrame].
+ *
+ * This function does not immediately rename the columns but instead selects columns to rename and
+ * returns a [RenameClause],
+ * which serves as an intermediate step.
+ * The [RenameClause] object provides methods to rename selected columns using:
+ * - [into(name)][RenameClause.into] - renames selected columns to the specified names.
+ * - [into { nameExpression }][RenameClause.into] - renames selected columns using a provided
+ * expression assuming column with its path and returning a new name.
+ * - [toCamelCase()][RenameClause.toCamelCase] - renames all selected columns to "camelCase".
+ *
+ * Each method returns a new [DataFrame] with the renamed columns.
+ *
+ * Check out [Grammar].
+ *
+ * @include [SelectingColumns.ColumnGroupsAndNestedColumnsMention]
+ *
+ * See [Selecting Columns][RenameSelectingOptions].
+ *
+ * For more information: {@include [DocumentationUrls.Rename]}
+ */
+internal interface RenameDocs {
+
+    /**
+     * {@comment Version of [SelectingColumns] with correctly filled in examples}
+     * @include [SelectingColumns] {@include [SetRenameOperationArg]}
+     */
+    interface RenameSelectingOptions
+
+    /**
+     * ## Rename Operation Grammar
+     * {@include [LineBreak]}
+     * {@include [DslGrammarLink]}
+     * {@include [LineBreak]}
+     *
+     * **[`rename`][rename]**`  { columnsSelector: `[`ColumnsSelector`][ColumnsSelector]`  }`
```

Comment:
Don't forget everything explicitly written in Kotlin code needs to be bold, so also the braces etc.

Reply:
I did the same for format: https://github.com/Kotlin/dataframe/blob/b4a00369b1de8163f88b3dd593518499e8bb8fbd/core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/format.kt#L144

(And we also describe it in the DslGrammar description)